### ISSUE TITLE
Stops working after waking up from suspend

### ISSUE THREAD

**Issue by BillyDM** [zzkW35/amdgpu-fan#2]
After waking up my computer from suspend, the amdgpu-fan service seems to stop working (the fans are spinning noticeably louder).

If I run:
```
sudo systemctl stop amdgpu-fan
sudo systemctl start amdgpu-fan
```

Then it works again. I didn't have this issue when the aur package still used the chestm007 repo. I wonder if changing `WantedBy=default.target` to `WantedBy=sysinit.target` in the service file might be causing the issue?

**Comment by BillyDM** [comment]
I tried manually setting it to `WantedBy=default.target` and disabling and re-enabling the service, but that didn't fix it.

**Comment by BillyDM** [comment]
Oh, I see now that you made this commit, so the issue probably lies there:

https://github.com/chestm007/amdgpu-fan/commit/82649eb4b396188855ebec19ca34a0c1c937daca

For reference, this is the fan curve I made in `etc/amdgpu-fan.yml`:

```
speed_matrix:  # -[temp(*C), speed(0-100%)]
- [0, 30]
- [70, 30]
- [75, 40]
- [80, 50]
- [85, 75]
- [90, 100]
```

**Comment by igo95862** [comment]
FYI there is a bug with Navi where fan speed fails to be red: https://gitlab.freedesktop.org/drm/amd/-/issues/1164

I would rather remove dependency on fan speed. Make the temperature drop be calculated by caching last temperature that the fan speed was set at and that last fan speed. Always write fan speed. ([there is a giant performance bump in only opening sysfs files once](https://www.kernel.org/doc/html/latest/filesystems/sysfs.html#reading-writing-attribute-data))

**Comment by roggif** [comment]
Hi, 
first of all thanks to both of you, for opening the issue and for suggesting a possible fix, but sadly I need to put this issue on hold for the moment. I'm currently away from my desktop PC (which is the only machine I own with AMD graphics) and will stay this way for quite some time (~1 month), this means that -obviously- I won't be able to test anything on practice. 

I **might** push something to the staging branch (nothing assured), but -again- it will be **untested** code.

Sorry for the inconvenience.

**Comment by igo95862** [comment]
@zzkW35 Would you be able to merge Pull Requests?

**Comment by roggif** [comment]
> @zzkW35 Would you be able to merge Pull Requests?

Yes, but only in the staging branch. I want to personally test the code before committing it to master.

**Comment by BillyDM** [comment]
That's okay, no rush. The old chestm007 repo on my system still works fine for now.

**Comment by roggif** [comment]
@BillyDM I just reverted that one commit you pointed out, can you test the staging branch and let me know if you keep facing the issue?

While you're at it, can you share your PC specs?

Thanks.

**Comment by BillyDM** [comment]
> @BillyDM I just reverted that one commit you pointed out, can you test the staging branch and let me know if you keep facing the issue?
> 
> While you're at it, can you share your PC specs?
> 
> Thanks.

I tried the staging branch and it did not fix the issue. So it seems that wasn't the problem.
